plugins/modules/iso.py

```diff
@@ -125,20 +126,27 @@ def ensure_present(module, rest_client):
     TaskTag.wait_task(rest_client, task_tag_create)
 
     # Uploading ISO image.
-    file_size = os.stat(module.params["source"]).st_size
-    with open(module.params["source"], "rb") as source_file:
-        rest_client.put_record(
-            endpoint="/rest/v1/ISO/%s/data" % iso_uuid,
-            payload=None,
-            check_mode=module.check_mode,
-            timeout=ISO_TIMEOUT_TIME,
-            binary_data=source_file,
-            headers={
-                "Content-Type": "application/octet-stream",
-                "Accept": "application/json",
-                "Content-Length": file_size,
-            },
+    try:
+        file_size = os.stat(module.params["source"]).st_size
+        with open(module.params["source"], "rb") as source_file:
+            rest_client.put_record(
+                endpoint="/rest/v1/ISO/%s/data" % iso_uuid,
+                payload=None,
+                check_mode=module.check_mode,
+                timeout=ISO_TIMEOUT_TIME,
+                binary_data=source_file,
+                headers={
+                    "Content-Type": "application/octet-stream",
+                    "Accept": "application/json",
+                    "Content-Length": file_size,
+                },
+            )
+    except FileNotFoundError:
+        raise errors.ScaleComputingError(
+            f"ISO file {module.params['source']} not found."
         )
+    except (json.JSONDecodeError, ValueError):
+        pass  # ISO API endpoint returns binary content.
```

Comment:
We can `pass` here, since the status code from response is acceptable.
Unacceptable status codes are raised in the client ... in case ISO upload was unsuccessful.

Reply:
I think so yes. This is the only module where we "pass" these json exceptions.
Other modules can handle them on their own if needed.
We should probably add these to other modules where we use `put`, but raise `ScaleError ` instead of pass.